src/StmtBase.h

```diff
@@ -144,16 +146,8 @@ class Stmt : public Obj {
     // hook body.
     virtual bool CouldReturn(bool ignore_break) const { return false; }
 
-    // Access to the original statement from which this one is derived,
-    // or this one if we don't have an original.  Returns a bare pointer
-    // rather than a StmtPtr to emphasize that the access is read-only.
-    const Stmt* Original() const { return original ? original->Original() : this; }
-
-    // Designate the given Stmt node as the original for this one.
-    void SetOriginal(StmtPtr _orig) {
-        if ( ! original )
-            original = std::move(_orig);
-    }
+    void SetLocation(const ObjPtr& orig_obj) { SetLocation(orig_obj.get()); }
+    void SetLocation(const Obj* orig_obj) { SetLocationInfo(orig_obj->GetLocationInfo()); }
```

Comment:
Similar as in Expr: I want to suggest we don't need these on the `Stmt` themselves given they are essentially `stmt->SetLocationInfo(other->GetLocationInfo())` which is just a bit more verbose and there aren't too many call-sites in `Stmt.cc`.

I realize there's extensive usage in script_opt, but you could have a free helper `set_location(Obj *, Obj *)`, too.

Reply:
Cool, I'm fine with streamlining this